mlir/lib/Bindings/Python/IRCore.cpp

```diff
@@ -450,6 +458,9 @@ class PyBlockList {
 
   PyBlock dunderGetItem(intptr_t index) {
     operation->checkValid();
+    if (index < 0) {
+      index += dunderLen();
+    }
     if (index < 0) {
       throw nb::index_error("attempt to access out of bounds block");
     }
```

Comment:
DCE?

Reply:
It can still be negative, right?

```
xs = [1, 2, 3]
xs[-42]
```

Reply:
Oh whoops duh. Ok

Reply:
Out-of-scope for your PR here, but could these exceptions be a bit more expressive by including the original index, the adjusted one, and the bounds in the error message?

Reply:
The test suite doesn't check OOB errors at the moment, I think.